yb-voyager/cmd/analyzeSchema.go

```diff
@@ -374,12 +376,54 @@ func checkStmtsUsingParser(sqlInfoArr []sqlInfo, fpath string, objType string) {
 			continue
 		}
 
-		if objType == TABLE {
-			createTableNode, ok := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CreateStmt)
-			if ok {
-				reportGeneratedStoredColumnTables(createTableNode, sqlStmtInfo, fpath)
-			}
+		createTableNode, isCreateTable := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CreateStmt)
+		alterTableNode, isAlterTable := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_AlterTableStmt)
+		createIndexNode, isCreateIndex := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_IndexStmt)
+
+		if objType == TABLE && isCreateTable {
+			reportGeneratedStoredColumnTables(createTableNode, sqlStmtInfo, fpath)
+		}
+		if isAlterTable {
+			reportAlterTableVariants(alterTableNode, sqlStmtInfo, fpath, objType)
+		}
+		if isCreateIndex {
+			reportCreateIndexStorageParameter(createIndexNode, sqlStmtInfo, fpath)
+		}
+
+	}
+}
+
+func reportCreateIndexStorageParameter(createIndexNode *pg_query.Node_IndexStmt, sqlStmtInfo sqlInfo, fpath string) {
+	indexName := createIndexNode.IndexStmt.GetIdxname()
+	summaryMap["INDEX"].invalidCount[sqlStmtInfo.objName] = true
+	if len(createIndexNode.IndexStmt.GetOptions()) > 0 {
+		reportCase(fpath, STORAGE_PARAMETERS_DDL_STMT_ISSUE, "https://github.com/yugabyte/yugabyte-db/issues/23467",
+			"Remove the storage parameters from the DDL", "INDEX", indexName, sqlStmtInfo.stmt)
+	}
+}
+
+func reportAlterTableVariants(alterTableNode *pg_query.Node_AlterTableStmt, sqlStmtInfo sqlInfo, fpath string, objType string) {
+	schemaName := alterTableNode.AlterTableStmt.Relation.Schemaname
+	tableName := alterTableNode.AlterTableStmt.Relation.Relname
+	fullyQualifiedName := lo.Ternary(schemaName != "", schemaName+"."+tableName, tableName)
+	if objType == "TABLE" {
+		// this will the list of items in the SET (attribute=value, ..)
+		setParameters := alterTableNode.AlterTableStmt.Cmds[0].GetAlterTableCmd().GetDef().GetList()
```

Comment:
GetList seems like a generic function. Are you sure that it's only going to get the set Parameters?

Reply:
Maybe adda link to documentation if any that would be helpful.

Reply:
> GetList seems like a generic function. Are you sure that it's only going to get the set Parameters?

Added a comment explanation for that but yes this List returns a list of items in the `list` field of the definition only in case of Set options is provided in the ALTER.